src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/RazorHtmlCSharpLanguageServerClient.cs

```diff
@@ -47,14 +57,19 @@ public RazorHtmlCSharpLanguageServerClient([ImportMany] IEnumerable<Lazy<IReques
             remove { }
         }
 
-        public Task<Connection> ActivateAsync(CancellationToken token)
+        public async Task<Connection> ActivateAsync(CancellationToken token)
         {
             var (clientStream, serverStream) = FullDuplexStream.CreatePair();
 
-            _languageServer = new RazorHtmlCSharpLanguageServer(serverStream, serverStream, _requestHandlers);
+            // Wait for logging infrastructure to initialize. This must be completed
+            // before we can start listening via Json RPC (as we must link the log hub
+            // trace source with Json RPC to facilitate structured logging / activity tracing).
+            await _loggerProvider.InitializeLoggerAsync().ConfigureAwait(false);
```

Comment:
Given this is dependent on the JsonRpc object should we just let the language server do this piece? Aka make RazorHtmlCSharpLanguageServer have a private ctor with a static public CreateAsync method that does this?

Reply:
Makes sense, updated!